src/java.desktop/macosx/native/libawt_lwawt/awt/CPrinterJob.m

```diff
@@ -664,15 +731,19 @@ static JNF_MEMBER_CACHE(jm_getDestinationFile, sjc_CPrinterJob,
 JNIEXPORT jboolean JNICALL Java_sun_lwawt_macosx_CPrinterJobDialog_showDialog
   (JNIEnv *env, jobject jthis)
 {
-    static JNF_CLASS_CACHE(jc_CPrinterJobDialog, "sun/lwawt/macosx/CPrinterJobDialog");
-    static JNF_MEMBER_CACHE(jm_pageable, jc_CPrinterJobDialog, "fPageable", "Ljava/awt/print/Pageable;");
+    DECLARE_CLASS_RETURN(jc_CPrinterJobDialog, "sun/lwawt/macosx/CPrinterJobDialog", NO);
+    DECLARE_FIELD_RETURN(jm_pageable, jc_CPrinterJobDialog, "fPageable", "Ljava/awt/print/Pageable;", NO);
 
     jboolean result = JNI_FALSE;
 JNF_COCOA_ENTER(env);
-    jobject printerJob = JNFGetObjectField(env, jthis, sjm_printerJob);
-    NSPrintInfo* printInfo = (NSPrintInfo*)jlong_to_ptr(JNFCallLongMethod(env, printerJob, sjm_getNSPrintInfo)); // AWT_THREADING Safe (known object)
-
-    jobject pageable = JNFGetObjectField(env, jthis, jm_pageable);
+    GET_CPRINTERDIALOG_METHOD_RETURN(NO);
+    jobject printerJob = (*env)->GetObjectField(env, jthis, sjm_printerJob);
+    if (printerJob == NULL) return NO;
+    GET_NSPRINTINFO_METHOD_RETURN(NO)
+    NSPrintInfo* printInfo = (NSPrintInfo*)jlong_to_ptr((*env)->CallLongMethod(env, printerJob, sjm_getNSPrintInfo)); // AWT_THREADING Safe (known object)
```

Comment:
CHECK_EXCEPTION?

Reply:
I've fixed a bunch of these but some - like this one - do not need CHECK_EXCEPTION.
We detect a null return value and return directly to Java and we are better to propagate any theoretical
exception in such cases.